lib/ruby_lsp/requests/inlay_hints.rb

```diff
@@ -0,0 +1,55 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    # ![Inlay hint demo](../../misc/inlay_hint.gif)
+    #
+    # [Inlay hints](https://microsoft.github.io/language-server-protocol/specification#textDocument_inlayHint)
+    # are labels added directly in the code that explicitly show the user something that might
+    # otherwise just be implied.
+    #
+    # # Example
+    #
+    # ```ruby
+    # begin
+    #   puts "do something that might raise"
+    # rescue # Label "StandardError" goes here as a bare rescue implies rescuing StandardError
+    #   puts "handle some rescue"
+    # end
+    # ```
+
+    class InlayHints < BaseRequest
+      sig { params(document: Document).void }
+      def initialize(document)
+        @hints = T.let([], T.untyped) # Released in LSP 3.17
+        super(document)
+      end
+
+      sig { override.returns(T.untyped) } # Released in LSP 3.17
+      def run
+        visit(@document.tree)
+        @hints
+      end
+
+      sig { params(node: T.nilable(SyntaxTree::Node)).void }
+      def visit(node)
+        return if node.nil?
+
+        if node.is_a?(SyntaxTree::Rescue) && node.exception.nil?
```

Comment:
Instead of checking the node type, we can just define `visit_rescue` and it'll only be invoked on rescue nodes.

Additionally, you can use the range parameter sent from the client to skip over rescue nodes that are not in the visible range.

Something like this
```ruby
sig { params(node: SyntaxTree::Rescue).void }
def visit_rescue(node)
  return unless node.exception.nil?

  loc = node.location
  return unless loc.start_line - 1 >= @range.dig(:start, :line) && loc.end_line - 1 <= @range.dig(:end, :line)

  @hints << ...
end
```